packages/apps/src/plugins/http/plugin.ts

```diff
@@ -217,15 +240,33 @@ export class HttpPlugin implements ISender {
     }
 
     const activity: Activity = req.body;
-    const token: IToken = authorization
-      ? new JsonWebToken(authorization)
-      : {
-          appId: '',
-          from: 'azure',
-          fromId: '',
-          serviceUrl: activity.serviceUrl || '',
-          isExpired: () => false,
-        };
+    let token: IToken;
+    if (this.serviceTokenValidator) {
+      if (!authorization) {
+        res.status(401).send('unauthorized no authorization header');
+        return;
+      }
+
+      // Use cached validator with per-request service URL validation
+      const validationResult = await this.serviceTokenValidator.validateAccessToken(authorization, activity.serviceUrl ? {
```

Comment:
I think we should stick to making this an optional configuration to avoid customers that don't want it not being able to disable it.

Reply:
sure, but why would a customer want to disable this?

Reply:
for example, what if this request was coming from webchat or some other service other than teams? There is no guarantee all providers of the activity protocol will use this form of auth.

Also, in the case of things like devtools/testing or even "I have protocol A and ill send it as an activity payload to you", you would likely want to disable this functionality. At the very least its an escape hatch and I typically modularize these kinds of things to give devs the power of choice, although I could see us having this enabled by default.

Reply:
Added `skip-auth` which optionally allows you to disable it. Also verified that the stuff works with web-chat, so i think this is safe to enable for all channels. But in case someone complains, they can pass in `skip-auth`.